src/Mvc/Mvc.Core/src/ApplicationParts/RelatedAssemblyAttribute.cs

```diff
@@ -16,7 +17,8 @@ namespace Microsoft.AspNetCore.Mvc.ApplicationParts
     [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
     public sealed class RelatedAssemblyAttribute : Attribute
     {
-        private static readonly Func<string, Assembly> AssemblyLoadFileDelegate = Assembly.LoadFile;
+        private static readonly Func<string, Assembly> LoadFromAssemblyPathDelegate =
+            AssemblyLoadContext.GetLoadContext(typeof(RelatedAssemblyAttribute).Assembly).LoadFromAssemblyPath;
```

Comment:
I would not do this - `RelatedAssemblyAttribute` is from Mvc.Core which is part of the framework, so pretty much always loaded into the default ALC. So this is the same as doing `AssemblyLoadContext.Default.LoadFromAssemblyPath`.
If in the future something loads the "main" assembly (the one with the related assemblies on it) into a secondary ALC this will likely be wrong and things might break. I think it should be `AssemblyLoadContext.GetLoadContext(assembly).LoadFromAssemblyPath` - where `assembly` is the "main" assembly.